pkg/await/core_service_test.go

```diff
@@ -99,31 +98,9 @@ func Test_Core_Service(t *testing.T) {
 			expectedError: &timeoutError{
 				object: serviceInput("default", "foo-4setj4y6"),
 				subErrors: []string{
-					"Service does not target any Pods. Selected Pods may not be ready, or " +
-						"field '.spec.selector' may not match labels on any Pods",
+					`Pods selected by Service "foo-4setj4y6" are not ready.`,
 					"Service was not allocated an IP address; does your cloud provider support this?"}},
 		},
-		{
-			description:  "Should fail if Endpoints have not initialized",
-			serviceInput: serviceInput,
-			do: func(services, endpoints chan watch.Event, settled chan struct{}, timeout chan time.Time) {
-				// API server passes initialized service back.
-				services <- watchAddedEvent(initializedService("default", "foo-4setj4y6"))
-
-				// Pass uninitialized endpoint objects. Mark them as settled.
-				endpoints <- watchAddedEvent(
-					uninitializedEndpoint("default", "foo-4setj4y6"))
-				settled <- struct{}{}
-
-				// Finally, time out.
-				timeout <- time.Now()
-			},
-			expectedError: &timeoutError{
-				object: initializedService("default", "foo-4setj4y6"),
-				subErrors: []string{
-					"Service does not target any Pods. Selected Pods may not be ready, or " +
-						"field '.spec.selector' may not match labels on any Pods"}},
```

Comment:
Is the progression the following? Endpoint created with `null` subsets -> controller picks up pods -> controller populates subsets?

If so, then this condition should cause ~all non-LoadBalancer services to immediately succeed, right?

Reply:
Hmm, maybe initially. I wonder if a short sleep would be sufficient? I'm not completely happy with this solution, but it's the only way I could think of to handle the 0-replica case.

Reply:
I have an idea for reworking this, but it would require more effort and testing. In short, I think we could check active ReplicaSets with matching selectors to determine how many replicas to expect. I don't think this will fit into the 1.0 timeframe, but I'll loop back to it soon after.

Reply:
We should do extensive testing on this. I am super leery now of changing await conditions without this.